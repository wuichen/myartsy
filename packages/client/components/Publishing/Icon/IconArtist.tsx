import React from "react"

export const IconArtist: React.SFC<{ color?: string }> = props => {
  return (
    <svg
      className="IconArtist"
      x="0px"
      y="0px"
      viewBox="0 0 14 18"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
        <g transform="translate(-1066.000000, -884.000000)" fill={props.color}>
          <g transform="translate(1066.000000, 884.000000)">
            <g transform="translate(0.000000, 9.000000)" stroke={props.color}>
              <rect stroke="none" x="0" y="0" width="14" height="9" />
            </g>
            <path d="M7,7 C8.93299662,7 10.5,5.43299662 10.5,3.5 C10.5,2.56042226 10.1297687,1.70731668 9.52723369,1.07861098 C8.89017133,0.413878211 7.99341888,0 7,0 C5.06700338,0 3.5,1.56700338 3.5,3.5 C3.5,5.43299662 5.06700338,7 7,7 Z" />
          </g>
        </g>
      </g>
    </svg>
  )
}

IconArtist.defaultProps = {
  color: "black",
}
