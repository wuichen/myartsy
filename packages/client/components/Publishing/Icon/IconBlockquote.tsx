import React from "react"

export const IconBlockquote: React.SFC<{ color?: string }> = props => {
  return (
    <svg
      className="IconBlockquote"
      viewBox="0 0 14.29 15.4"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fill={props.color}
        d="M0,3.29A3.6,3.6,0,0,1,2.83,0,.36.36,0,0,1,3,.51C2.11.92,1.16,1.6,1.16,2.59A1.33,1.33,0,0,0,2.37,3.95.64.64,0,0,1,3,4.62c0,.36-.22,1-1.16,1A2.18,2.18,0,0,1,0,3.29Zm4.09,0A3.6,3.6,0,0,1,6.92,0a.36.36,0,0,1,.17.51C6.2.92,5.25,1.6,5.25,2.59A1.33,1.33,0,0,0,6.46,3.95a.64.64,0,0,1,.65.68c0,.36-.22,1-1.16,1A2.18,2.18,0,0,1,4.09,3.29Z"
      />
      <path
        fill={props.color}
        d="M10.2,12.11A3.6,3.6,0,0,1,7.37,15.4a.36.36,0,0,1-.17-.51C8.09,14.48,9,13.81,9,12.81a1.33,1.33,0,0,0-1.21-1.36.64.64,0,0,1-.65-.68c0-.36.22-1,1.16-1A2.18,2.18,0,0,1,10.2,12.11Zm4.09,0a3.6,3.6,0,0,1-2.83,3.29.36.36,0,0,1-.17-.51c.9-.41,1.84-1.09,1.84-2.08a1.33,1.33,0,0,0-1.21-1.36.64.64,0,0,1-.65-.68c0-.36.22-1,1.16-1A2.18,2.18,0,0,1,14.29,12.11Z"
      />
    </svg>
  )
}

IconBlockquote.defaultProps = {
  color: "black",
}
