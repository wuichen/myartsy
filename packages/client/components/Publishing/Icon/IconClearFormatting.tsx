import React from "react"

export const IconClearFormatting: React.SFC<{ color?: string }> = props => {
  return (
    <svg
      className="IconClearFormatting"
      x="0px"
      y="0px"
      viewBox="0 0 15.5 15.8"
      enableBackground="new 0 0 15.5 15.8"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          fill={props.color}
          d="M10.3,0.4c0.7,0,0.9-0.1,1.2-0.4l0.4,0.1c-0.3,0.5-0.8,1.8-0.9,2.3c-0.1,0.1-0.4,0.1-0.5,0c0.1-0.6,0.1-1.1-0.2-1.3 	C10,1.1,9.6,1,9.1,1H8.3C7.9,1,7.9,1,7.7,1.6L5.6,8.2C5.2,9.5,5.2,9.8,6,9.9l0.6,0.1c0.1,0.1,0,0.4-0.1,0.4c-1,0-1.6,0-2.3,0 c-0.7,0-1.3,0-2.1,0c-0.1-0.1-0.1-0.4,0-0.4l0.6-0.1C3.4,9.8,3.5,9.6,4,8.2L6,1.6C6.2,1,6.2,1,5.7,1H4.9C4.4,1,4,1.1,3.6,1.3 C3.2,1.5,2.9,1.9,2.5,2.5C2.3,2.6,2.1,2.5,2,2.3c0.5-0.8,0.9-1.7,1.1-2.2C3.2,0,3.4,0,3.5,0c0,0.3,0.2,0.4,1.1,0.4H10.3z"
        />
        <polyline fill={props.color} points="8,14.2 8,13.2 0,13.2 0,14.2" />
        <polygon
          fill={props.color}
          points="15.2,11.6 14.5,10.9 12.9,12.5 11.3,10.9 10.6,11.6 12.2,13.2 10.6,14.8 11.3,15.5 12.9,13.9 14.5,15.5 15.2,14.8 13.6,13.2"
        />
      </g>
    </svg>
  )
}

IconClearFormatting.defaultProps = {
  color: "black",
}
