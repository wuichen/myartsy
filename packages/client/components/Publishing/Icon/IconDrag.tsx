import React from "react"

interface IconDragProps {
  color?: string
  background?: string
}

export const IconDrag: React.SFC<IconDragProps> = ({
  color = "white",
  background = "black",
}) => {
  return (
    <svg
      className="IconDrag"
      x="0px"
      y="0px"
      viewBox="0 0 30 30"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g transform="translate(-1062.000000, -935.000000)">
        <g transform="translate(1062.000000, 934.000000)">
          <g transform="translate(0.785741, 0.701007)">
            <circle fill={background} cx="14.5" cy="15" r="14.5" />
            <g
              transform="translate(10.000000, 6.500000)"
              strokeWidth="2"
              stroke={color}
            >
              <polyline
                fill={background}
                transform="translate(4.500000, 4.923718) rotate(-45.000000) translate(-4.500000, -4.923718)"
                points="1.5 1.92371784 7.5 1.92371784 7.5 7.92371784"
              />
              <polyline
                fill={background}
                transform="translate(4.500000, 12.227216) rotate(-225.000000) translate(-4.500000, -12.227216)"
                points="1.5 9.2272156 7.5 9.2272156 7.5 15.2272156"
              />
              <path d="M4.5,1.98457492 L4.5,16.0089159" />
            </g>
          </g>
        </g>
      </g>
    </svg>
  )
}
