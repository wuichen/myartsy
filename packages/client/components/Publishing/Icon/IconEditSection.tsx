import React from "react"

export const IconEditSection: React.SFC<{
  fill?: string
  isClosing?: boolean
  width?: string
}> = ({ isClosing, width = "45px", fill = "black" }) => {
  return (
    <svg
      className="edit-section"
      x="0px"
      y="0px"
      width={width}
      viewBox="0 0 1000 1000"
      enableBackground="0 0 1000 1000"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        x="158"
        y="304"
        fill="#FFF"
        strokeMiterlimit="10"
        width="436"
        height="399"
      />
      {isClosing ? (
        <path
          fill={fill}
          d="M379.318,132.208c-194.157,0-355.33,163.644-355.33,359.888s161.173,359.888,355.33,359.888 c42.962,0,213.011-0.984,571.915-359.888C592.329,133.192,422.28,132.208,379.318,132.208z M527.73,451.989v71.963l-311.841,0 v-71.963L527.73,451.989z"
        />
      ) : (
        <path
          fill={fill}
          d="M379.318,132.208c-194.157,0-355.33,163.644-355.33,359.888s161.173,359.888,355.33,359.888 c42.962,0,213.011-0.984,571.915-359.888C592.329,133.192,422.28,132.208,379.318,132.208z M407.791,523.952V643.89h-71.963V523.952 H215.889v-71.963h119.939V332.05h71.963v119.939H527.73v71.963H407.791z"
        />
      )}
    </svg>
  )
}
