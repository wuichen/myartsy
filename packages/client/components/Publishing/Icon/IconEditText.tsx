import React from "react"

export const IconEditText: React.SFC<{ width?: string }> = ({
  width = "45px",
}) => {
  return (
    <svg
      className="edit-text"
      x="0px"
      y="0px"
      width={width}
      viewBox="0 0 56.769 29.866"
      enableBackground="new 0 0 56.769 29.866"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect x="1.953" y="13.344" width="52" height="3.447" />
      <rect x="1.953" y="24.348" width="33.224" height="3.447" />
      <rect x="1.953" y="2.475" width="52" height="3.447" />
    </svg>
  )
}
