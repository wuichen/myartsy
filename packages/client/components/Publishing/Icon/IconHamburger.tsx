import React, { Component } from "react"

export class IconHamburger extends Component<any, null> {
  static defaultProps = {
    color: "black",
    onClick: x => x,
  }

  render() {
    return (
      <svg
        onClick={this.props.onClick}
        fill={this.props.color}
        width="32px"
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 32 32"
        enableBackground="new 0 0 32 32"
      >
        <rect x="6" y="22" width="20" height="2" />
        <rect x="6" y="15" width="20" height="2" />
        <rect x="6" y="8" width="20" height="2" />
      </svg>
    )
  }
}
