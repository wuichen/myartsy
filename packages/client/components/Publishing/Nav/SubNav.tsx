import { avantgarde } from "Assets/Fonts"
import React from "react"
import styled from "styled-components"
import StyledButton from "../../Buttons/Default"
import InvertedButton from "../../Buttons/Inverted"
import { pMedia } from "../../Helpers"
import Icon from "../../Icon"
import { MobileMenu } from "../../Nav/MobileMenu"
import { IconHamburger } from "../Icon/IconHamburger"

interface State {
  mobileNavIsOpen?: boolean
}

export class SubNav extends React.Component<null, State> {
  constructor(props) {
    super(props)
    this.state = {
      mobileNavIsOpen: false,
    }
  }

  toggleMobileNav = () => {
    this.setState({
      mobileNavIsOpen: !this.state.mobileNavIsOpen,
    })
  }

  render() {
    return (
      <SubNavContainer>
        <Link className="mlh-login" href="/log_in">
          <Login>Login</Login>
        </Link>
        <Link className="mlh-signup" href="/sign_up">
          <Signup>Signup</Signup>
        </Link>
        <MobileNavToggle>
          {this.state.mobileNavIsOpen ? (
            <Icon name="close" color="white" onClick={this.toggleMobileNav} />
          ) : (
            <IconHamburger color="white" onClick={this.toggleMobileNav} />
          )}
        </MobileNavToggle>
        {this.state.mobileNavIsOpen && <MobileMenu navHeight="46px" />}
      </SubNavContainer>
    )
  }
}

const Login = styled(InvertedButton)`
  ${avantgarde("s11")};
  height: 30px;
  width: 80px;
  border-radius: 2px;
  margin: 8px;
`
const Signup = styled(StyledButton)`
  ${avantgarde("s11")};
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;
  margin: 8px;
`
const Link = styled.a`
  text-decoration: none;
`
const MobileNavToggle = styled.div`
  display: none;
  margin-right: 5px;
  max-width: 32px;
`
const SubNavContainer = styled.div`
  display: flex;
  height: 100%;
  ${pMedia.sm`
    ${Login}, ${Signup} {
      display: none;
    }
    ${MobileNavToggle} {
      display: flex;
      align-items: center;
    }
  `};
`
