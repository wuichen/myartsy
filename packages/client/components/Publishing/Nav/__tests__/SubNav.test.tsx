import { IconHamburger } from "Components/Publishing/Icon/IconHamburger"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import Icon from "../../../Icon"
import { SubNav } from "../SubNav"

describe("SubNav", () => {
  it("renders SubNav", () => {
    // @ts-ignore
    const buttons = renderer.create(<SubNav />) as any
    expect(buttons).toMatchSnapshot()
  })
  it("toggles the mobile nav", () => {
    // @ts-ignore
    const nav = mount(<SubNav />) as any
    nav.find(IconHamburger).simulate("click")
    expect(nav.state().mobileNavIsOpen).toBe(true)
    nav.find(Icon).simulate("click")
    expect(nav.state().mobileNavIsOpen).toBe(false)
  })
})
