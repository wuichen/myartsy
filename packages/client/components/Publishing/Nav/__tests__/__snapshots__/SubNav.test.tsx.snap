// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNav renders SubNav 1`] = `
.c3 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 13px;
  line-height: 1.65em;
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color 0.25s,color 0.25s;
  transition: background-color 0.25s,color 0.25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 2px;
  background: black;
  color: white;
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  height: 30px;
  width: 80px;
  border-radius: 2px;
  margin: 8px;
}

.c3:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c3:hover:not(:disabled) {
  background: #6e1fff;
}

.c5 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 13px;
  line-height: 1.65em;
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color 0.25s,color 0.25s;
  transition: background-color 0.25s,color 0.25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 2px;
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;
  margin: 8px;
}

.c5:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  display: none;
  margin-right: 5px;
  max-width: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

@media (max-width:720px) {
  .c0 .c2,
  .c0 .c4 {
    display: none;
  }

  .c0 .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div
  className="c0"
>
  <a
    className="mlh-login c1"
    href="/log_in"
  >
    <button
      className="c2 c3"
    >
      <span>
        Login
      </span>
    </button>
  </a>
  <a
    className="mlh-signup c1"
    href="/sign_up"
  >
    <button
      className="c4 c5"
    >
      <span>
        Signup
      </span>
    </button>
  </a>
  <div
    className="c6 c7"
  >
    <svg
      enableBackground="new 0 0 32 32"
      fill="white"
      onClick={[Function]}
      version="1.1"
      viewBox="0 0 32 32"
      width="32px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="2"
        width="20"
        x="6"
        y="22"
      />
      <rect
        height="2"
        width="20"
        x="6"
        y="15"
      />
      <rect
        height="2"
        width="20"
        x="6"
        y="8"
      />
    </svg>
  </div>
</div>
`;
